components/datetime/src/fieldset.rs

```diff
@@ -21,26 +24,70 @@ use icu_calendar::{
 use icu_provider::marker::NeverMarker;
 use icu_timezone::{TimeZoneBcp47Id, UtcOffset, ZoneVariant};
 
+/// Enumerations over field sets.
+pub mod dynamic {
+    // TODO: Rename to `pub mod enums`
+    pub use crate::dynamic::*;
+}
+
 #[cfg(doc)]
 use icu_timezone::TimeZoneInfo;
 
 /// Maps the token `yes` to the given ident
 macro_rules! yes_to {
-    ($any:ident, yes) => {
+    ($any:expr, yes) => {
         $any
     };
     () => {
         unreachable!() // prevent bugs
     };
 }
 
+macro_rules! yes_or {
+    ($fallback:expr, $actual:expr) => {
+        $actual
+    };
+    ($fallback:expr,) => {
+        $fallback
+    };
+}
+
+macro_rules! ternary {
+    ($present:expr, $missing:expr, yes) => {
```

Comment:
nit: should we match `no`?

Reply:
We match either "present" or "missing", i.e. "yes" and ""